### DISCUSSION TITLE
Listing all the available packages in a machine-friendly fashion?

### DISCUSSION THREAD

**Discussion by VioletGiraffe** [microsoft/vcpkg#19161] [Ideas]
I'm working on a simple GUI for vcpkg, just to make my life easier (not that vcpkg is difficult to use). I want to list all the available packages, and I have a problem parsing `vcpkg search --x-full-desc` output, as well as `vcpkg search`. It's because of the lack of unambiguous delimiters, and also because some packages are missing version number. Is there a way to get this list in a more computer-friendly form? Perhaps, parse some file within `vcpkg` installation folder instead of parsing `vcpkg search` output?

**Comment by ras0219-msft** [comment]
Thanks for the suggestion!

I'd love to have a mode that outputs in a machine friendly format (`--xml`?). I think this would be better than trying to "track" vcpkg's parser for the control files.

I'd be happy to review a PR adding this!

**Comment by cmpute** [comment]
You could have a look at [my progress here](https://github.com/cmpute/vcpkg-gui). Directly reading the ports directory is quite simple.

**Comment by VioletGiraffe** [comment]
@cmpute : very nice work, that makes my project obsolete. I'll just use your GUI :)
My mistake was looking for a single file that describes the repository state; I didn't notice the "ports" folder.

**Comment by MikeGitb** [comment]
I think it would still be useful to have a integrated function that e.g. produces a json output listing both available packages as well as installed packages (installed packages might have a different version than the port files). That way, not everyone who wants to build tools on top of vcpkg has to search and parse the ports and packages directories.

**Comment by henrik-jensen** [comment]
> You could have a look at [my progress here](https://github.com/cmpute/vcpkg-gui). Directly reading the ports directory is quite simple.

@cmpute - ~~Could you point to where you attack this problem in your repo?~~

**Edit:**
Oh sorry, forget it, - found it.
It's the parser in [Port.cs](https://github.com/cmpute/vcpkg-gui/blob/master/UserInterface/Port.cs) ( obviously, d'oh :-) ) you are refering to.

**Comment by JackBoosY** [comment]
Thanks for @VioletGiraffe  @cmpute , I have the same idea too.

**Comment by NancyLi1013** [comment]
This issue hasn’t been updated in a year; if it is still an issue, please reopen.

**Comment by MikeGitb** [comment]
Best I can tell, this is still not implemented please reopen.

**Comment by VioletGiraffe** [comment]
I concur, but I also understand that this is a low priority task for vcpkg maintainers.

**Comment by PhoebeHui** [comment]
I noticed there is an option '--x-json' for listing  libraries in JSON format, but it's still in experimental.

```
  --x-json                        (experimental) List libraries in JSON format
```
